core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -497,45 +497,61 @@ sealed trait AsyncRequest {
   type Response <: AsyncResponse
   def path: String
   def ctx: Option[Any]
+  def registerWatch: Boolean
```

Comment:
This seems unnecessary given where we use this a few lines above.

Reply:
I agree. I have removed it.